nima/tests/integration/webserver/webserver/src/test/java/io/helidon/nima/tests/integration/server/ErrorHandlingWithOutputStreamTest.java

```diff
@@ -0,0 +1,145 @@
+                    os.write("|writeTwice".getBytes(StandardCharsets.UTF_8));
+                    throw new CustomException();
+                })
+                .get("get-outputStream-writeFlushThenError", (req, res) -> {
+                    OutputStream os = res.outputStream();
+                    os.write("writeOnce".getBytes(StandardCharsets.UTF_8));
+                    os.flush();
+                    throw new CustomException();
+                })
+                .get((req, res) -> res.send("ok"));
+    }
+
+    @Test
+    void testOk() {
+        var response = client.get().request();
+
+        assertThat(response.status(), is(Http.Status.OK_200));
+        assertThat(response.entity().as(String.class), is("ok"));
+    }
+
+    @Test
+    void testGetOutputStreamThenError_expect_CustomErrorHandlerMessage() {
+        var response = client.get("/get-outputStream").request();
+
+        assertThat(response.status(), is(Http.Status.I_AM_A_TEAPOT_418));
+        assertThat(response.entity().as(String.class), is("CustomErrorContent"));
+    }
+
+    @Test
+    void testGetOutputStreamWriteOnceThenError_expect_CustomErrorHandlerMessage() {
+        var response = client.get("/get-outputStream-writeOnceThenError").request();
+
+        assertThat(response.status(), is(Http.Status.I_AM_A_TEAPOT_418));
+        assertThat(response.entity().as(String.class), is("CustomErrorContent"));
+    }
+
+    @Test
+    void testGetOutputStreamWriteTwiceThenError_expect_CloseConnectionExceptionAndContentFromMainHandler() {
+        Headers headers;
+        try (Http1ClientResponse response = client.method(Http.Method.GET)
+                .uri("/get-outputStream-writeTwiceThenError")
+                .request()) {
+            assertThat(response.status(), is(Http.Status.OK_200));
+            assertThrows(DataReader.InsufficientDataAvailableException.class, () -> response.entity().as(String.class));
+            headers = response.headers();
+        }
+        // FIXME: It this supposed to be a (trailing) header for connection close ??
+        // assertThat(headers, hasHeader(Http.HeaderValues.CONNECTION_CLOSE));
+    }
+
+    @Test
+    void testGetOutputStreamWriteFlushThenError_expect_CloseConnectionExceptionAndContentFromMainHandler() {
+        Headers headers;
+        try (Http1ClientResponse response = client.method(Http.Method.GET)
+                .uri("/get-outputStream-writeFlushThenError")
+                .request()) {
+
+            // Getting (because Http1ServerResponse BlockingOutputStream.flush() is not overwritten
+            // and doesn't write the chunked firstBuffer - looks like a bug here for BlockingOutputStream.flush()
+            assertThat(response.status(), is(Http.Status.I_AM_A_TEAPOT_418));
+            assertThat(response.entity().as(String.class), is("CustomErrorContent"));
+
+            // FIXME: Expecting the below, looks like bug with flush() in Http1ServerResponse BlockingOutputStream
```

Comment:
To me this looks like a separate bug here with BlockingOutputStream.flush() - when there is only one write() and firstBuffer is set but not sent ... then flush() really should send that firstBuffer content but it doesn't